pkg/networkutils/network.go

```diff
@@ -243,10 +243,16 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, vpcCIDRs []*string,
 		primaryIntfRPFilter := "net/ipv4/conf/" + primaryIntf + "/rp_filter"
 		const rpFilterLoose = "2"
 
-		log.Debugf("Setting RPF for primary interface: %s", primaryIntfRPFilter)
-		err = n.procSys.Set(primaryIntfRPFilter, rpFilterLoose)
-		if err != nil {
-			return errors.Wrapf(err, "failed to configure %s RPF check", primaryIntf)
+		if n.procSys.IsPathWriteAccessible(primaryIntfRPFilter) {
+			// Setting RPF will be removed from aws-node when we bump our major version.
+			log.Debugf("Setting RPF for primary interface: %s", primaryIntfRPFilter)
+			err = n.procSys.Set(primaryIntfRPFilter, rpFilterLoose)
+			if err != nil {
+				return errors.Wrapf(err, "failed to configure %s RPF check", primaryIntf)
+		    }
+		} else {
+			// if aws-node run as un-privileged pods then /proc will be mounted as read only.
```

Comment:
Nit: `// If aws-node is running as an un-privileged pod, /proc will be mounted as read only.`